dask_expr/tests/test_merge.py

```diff
@@ -206,3 +207,26 @@ def test_merge_combine_similar(npartitions_left, npartitions_right):
     expected["new"] = expected.b + expected.c
     expected = expected.groupby(["a", "e", "x"]).new.sum()
     assert_eq(query, expected)
+
+
+def test_merge_combine_similar_intermediate_projections():
```

Comment:
Is there a reproducer for the behavior being targeted by this PR? This test seems to pass on `main` for me.

Reply:
Yeah the bug was blocked by the additional layer in #333 but shows up now